oat-sa/extension-tao-client-restrict
# Feature/ACTP-382/Import script

Create scripts and services to import browsers and operating systems (OS) from seed file.
 
Related to: [https://oat-sa.atlassian.net/browse/ACTP-382](https://oat-sa.atlassian.net/browse/ACTP-382)
 
#### Changes:
- added scripts to process and prepare data (browsers, OS) for import;
- added services to import processed data;
  
#### How to test:
1. add script `oat\\taoClientRestrict\\scripts\\tools\\import\\ImportScript` to the `postInstall` section in your seed file;
2. set `--list` parameter. It should be an object of JSON values;
3. set `--type` parameter. Available types:
     - `browser`;
     - `os`;
3. run `php tao/scripts/taoSetup.php [SEED FILE] -vvvv`;
4. go to the `Web Browsers Lists`/`OS Lists` and check installed instances.

Example:
```
"postInstall": [
    {
        "class": "oat\\taoClientRestrict\\scripts\\tools\\import\\ImportScript",
        "params": [
            "--list", [
                {
                    "classMap": [
                        "Test class"
                    ],
                    "label": "Chrome browser",
                    "name": "Chrome",
                    "version": "1.2.3"
                }
            ],
            "--type", "browser"
        ]
    },
    {
        "class": "oat\\taoClientRestrict\\scripts\\tools\\import\\ImportScript",
        "params": [
            "--list", [
                {
                    "classMap": [
                        "Test class"
                    ],
                    "label": "OS X operating system",
                    "name": "OS X",
                    "version": "3.2.1"
                }
            ],
            "--type", "os"
        ]
    }
]
```

#### List parameters description
| Parameter  | Description  | Required |
| ----------- | ------------ | --------- |
| classMap | Will generate the necessary folder structure from the root class | false |
| label | Label for browser/OS | true |
| name | Browser/OS name | true |
| version | Browser/OS version | true |
